src/ui/component/stepper/stepperProgress/stepperProgress.tsx

```diff
@@ -0,0 +1,73 @@
+import type { FC } from 'react'
+import type { ProgressBarTransition } from './progressBar/progressBar'
+import { ProgressBar } from './progressBar/progressBar'
+import './stepperProgress.scss'
+
+export type StepId = string
+export type StepStatus = 'error' | 'disabled' | 'complete' | 'incomplete'
+
+export type StepData = {
+  id: StepId
+  order: number
+  status: StepStatus
+}
+
+type StepperProgressProps = {
+  stepsData: StepData[]
+  activeStepId: StepId
+  previousActiveStepId: StepId
+}
+
+const getProgressBarTransition = ({
+  stepOrder,
+  activeStepOrder,
+  previousActiveStepOrder
+}: {
+  stepOrder: number
+  activeStepOrder: number
+  previousActiveStepOrder: number
+}): ProgressBarTransition => {
+  if (stepOrder === activeStepOrder && previousActiveStepOrder < stepOrder) {
+    return 'forward'
+  }
+  if (stepOrder === previousActiveStepOrder && activeStepOrder < stepOrder) {
+    return 'backward'
+  }
+  return 'none'
+}
+
+const findStepData = (stepsData: StepData[], stepId: StepId): StepData => {
+  const stepData = stepsData.find(({ id }) => id === stepId)
+  if (!stepData) {
+    throw new Error(`Step data not found for step id: ${stepId}`)
```

Comment:
Ok, we can throw an error if this rule is not respected and that will be caught by the `ErrorBoundary`.

On another side, we could prevent the app from crashing and handle a fallback behaviour that will be displayed by the `progressBar`.

Could be discussed 😌

Reply:
If we are using this function in the contexte of the stepper, it won't fail. It was mostly done to make stepper components more readable and avoiding to rely on the `as` keyword.

We can arbitrary fallback on the first step for instance. What do you think? Thanks

Reply:
Yes, I'll prefer to handle fallback behavior over throwing, thanks 😌